src/lib.rs

```diff
@@ -58,18 +66,20 @@ impl<W, T> PistonWindow<W, T>
         use piston::event::Events;
         use piston::window::{ OpenGLWindow, Window };
 
-        let (mut device, mut factory) = gfx_device_gl::create(|s| window.borrow_mut().get_proc_address(s));
+        let mut device = gfx_device_gl::Device::new(|s| window.borrow_mut().get_proc_address(s));
+        let mut factory = device.spawn_factory();
```

Comment:
@mitchmindtree `draw_3d` is the only function where factory was really used (as a part of `Canvas`). The general idea behind multiple factories is that anything that manages its own resources should just carry its own factory (thus `draw*` is not supposed to use the factory parameter). `GlyphCash` is a good example for it, it just needs to be updated.